packages/cxl-ui/src/components/cxl-jw-player/mixins/BaseMixin.js

```diff
@@ -0,0 +1,235 @@
+
+export function BaseMixin(BaseClass) {
+  class Mixin extends BaseClass {
+    _boundOnTimeListener;
+
+    _chapters;
+
+    _jwPlayer;
+
+    _jwPlayerContainer;
+
+    // Device Detector media query.
+    _wideMediaQuery = '(min-width: 750px)';
+
+    @property({ attribute: 'api-secret', type: String }) apiSecret = '';
+
+    @property({ attribute: 'is-public', type: Boolean }) isPublic;
+
+    @property({ attribute: 'library-id', type: String }) libraryId;
+
+    @property({ attribute: 'library-source', type: String }) librarySource;
+
+    @property({ attribute: 'media-id', type: String }) mediaId;
+
+    @property({ attribute: 'media-source', type: String }) mediaSource;
+
+    @property({ attribute: 'playlist-id', type: String }) playlistId;
+
+    @property({ attribute: 'playlist-source', type: String }) playlistSource;
+
+    // MediaQueryController.
+    @property({ type: Boolean, reflect: true })
+    wide;
+
+    constructor() {
+      super();
+
+      this.addController(
+        new MediaQueryController(this._wideMediaQuery, (matches) => {
+          this.wide = matches;
+        })
+      );
+    }
+
+    async firstUpdated(_changedProperties) {
+      await super.firstUpdated(_changedProperties);
+
+      await this._beforeSetup();
+      this._setup();
+    }
+
+    updated(_changedProperties) {
+      super.updated(_changedProperties);
+      if (_changedProperties.has('captions') || _changedProperties.has('mediaId')) {
+        // this._setup();
+      }
+    }
+
+    get _scriptUrl() {
+      if (!this.libraryId && !this.librarySource) return false;
+
+      let scriptUrl;
+
+      if (this.libraryId) {
+        if (this.isPublic) {
+          scriptUrl = `https://content.jwplatform.com/libraries/${this.libraryId}.js`;
+        } else {
+          scriptUrl = this.__signedURL(`libraries/${this.libraryId}.js`);
+        }
+      }
+
+      if (this.librarySource) {
+        scriptUrl = this.librarySource;
+      }
+
+      return scriptUrl;
+    }
+
+    _addStyle(style) {
+      const el = document.createElement('style');
+      render(style, el);
+      this.appendChild(el);
+    }
+
+    // eslint-disable-next-line class-methods-use-this, no-empty-function
+    _beforeSetup() {}
+
+    async _getChapters() {
+      const playlistItem = this._jwPlayer.getPlaylistItem();
+      const chapters = playlistItem.tracks.filter((track) => track.kind === 'chapters');
+
+      if (chapters.length === 0) {
+        return [];
+      }
+
+      const { file } = chapters.length > 0 ? chapters[0] : '';
+      const response = await (await fetch(file)).text();
+
+      return parseSync(response);
+    }
+
+    async _getMedia() {
+      if (!this.mediaId && !this.mediaSource) return false;
+
+      let response;
+
+      if (this.mediaId) {
+        if (this.isPublic) {
+          response = await fetch(`https://cdn.jwplayer.com/v2/media/${this.mediaId}`);
```

Comment:
`https://cdn.jwplayer.com` is repeated in the code multiple times. It should be defined as constant in the one place and used across the component